src/hotspot/share/gc/shenandoah/shenandoahFreeSet.cpp

```diff
@@ -288,112 +319,46 @@ HeapWord* ShenandoahFreeSet::try_allocate_in(ShenandoahHeapRegion* r, Shenandoah
   // req.size() is in words, r->free() is in bytes.
   if (ShenandoahElasticTLAB && req.is_lab_alloc()) {
     if (req.type() == ShenandoahAllocRequest::_alloc_plab) {
-      assert(_heap->mode()->is_generational(), "PLABs are only for generational mode");
       // Need to assure that plabs are aligned on multiple of card region.
-      size_t free = r->free();
-      // e.g. card_size is 512, card_shift is 9, min_fill_size() is 8
-      //      free is 514
-      //      usable_free is 512, which is decreased to 0
-      size_t usable_free = (free / CardTable::card_size()) << CardTable::card_shift();
-      if ((free != usable_free) && (free - usable_free < ShenandoahHeap::min_fill_size() * HeapWordSize)) {
-        // We'll have to add another card's memory to the padding
-        if (usable_free >= CardTable::card_size()) {
-          usable_free -= CardTable::card_size();
-        } else {
-          assert(usable_free == 0, "usable_free is a multiple of card_size and card_size > min_fill_size");
-        }
-      }
-      free /= HeapWordSize;
-      usable_free /= HeapWordSize;
-      size_t remnant = size % CardTable::card_size_in_words();
-      if (remnant > 0) {
-        // Since we have Elastic TLABs, align size up.  This is consistent with aligning min_size up.
-        size = size - remnant + CardTable::card_size_in_words();
-      }
+      // Since we have Elastic TLABs, align size up.  This is consistent with aligning min_size up.
```

Comment:
I's just say:
// Since we are using ElasticTLABs, we align the allocation up to card sizes/boundaries.